From d06b7705a4a58214e101f9d327fa00d25e2359d1 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Thu, 2 Apr 2015 09:56:10 +0800
Subject: [PATCH] MLK-10494: ASoC: imx_wm8960: one channel is mute when record
 2 channel sound

There is only one channel input in hardware connection. Route it to right
channel, then all 2 channel has sound.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 sound/soc/fsl/imx-wm8960.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sound/soc/fsl/imx-wm8960.c b/sound/soc/fsl/imx-wm8960.c
index e411142..0151796 100644
--- a/sound/soc/fsl/imx-wm8960.c
+++ b/sound/soc/fsl/imx-wm8960.c
@@ -106,6 +106,12 @@ static int imx_hifi_hw_params(struct snd_pcm_substream *substream,
 	snd_soc_update_bits(codec_dai->codec, WM8960_ADDCTL4, 3<<2, 2<<2);
 	snd_soc_update_bits(codec_dai->codec, WM8960_ADDCTL1, 1, 1);
 
+	/*
+	 * As the hardware only connect the input for left channel, we need
+	 * to route it for right channel.
+	 */
+	snd_soc_update_bits(codec_dai->codec, WM8960_ADDCTL1, 3<<2, 1<<2);
+
 	if (!data->is_codec_master) {
 		ret = snd_soc_dai_set_sysclk(cpu_dai, 0, 0, SND_SOC_CLOCK_OUT);
 		if (ret) {
-- 
1.8.0

